roboflow/core/version.py

```diff
@@ -136,6 +140,13 @@ def __init__(
             else:
                 self.model = None
 
+            # check if version has a model
+            if version_info.status_code == 200:
+                version_info = version_info.json()["version"]
+
+                if not bool(version_info["models"]):
+                    self.model = None
```

Comment:
This implementation removes any prior `model` value if there's no model of this version. If that's correct, I suggest moving this up, approx. to line 105, and returning early. No reason to instantiate `model` object if it will be reset afterwards.